assets/js/base/components/payment-methods/saved-payment-method-options.js

```diff
@@ -167,7 +208,7 @@ const SavedPaymentMethodOptions = ( { onSelect } ) => {
 
 	// In the editor, show `Use a new payment method` option as selected.
 	const selectedOption = isEditor ? '0' : selectedToken + '';
-	return currentOptions.current.length > 0 ? (
+	return currentOptions.current.length > 1 ? (
```

Comment:
That was part of the refactor in 73a19d1dbcee0f6d23cb17d1f826948dc461552a (see [related lines of code](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3134/files#diff-40dcd5cbcf22a46ff3ce6512834ba7f8L134-L144)) in order to make it easier to understand. Thinking about it twice, I should have broken it in two different PRs in order to make it easier to review. Let me know if you want me to split it.

The idea behind the refactor was to remove the `if` clause from the effect and the `push` to the array stored in the ref. I was having a hard time understanding how they worked and felt I was simplifying it. But I might have achieved the opposite. :sweat_smile: 

I did another try to simplify this (e3ca65d3fac7dfbb09a7003bea368177ce8cf523), basically instead of appending the default option inside the effect, we append it in the render.